### PR TITLE
Hadren styles for interactive elements in Checkout block

### LINKED ISSUE #50954: Be more aggressive in resetting panel toggles in Cart/Checkout block
Cart/Checkout uses a Panel component that uses a button to expand/collapse the section. While a button is the right a11y component to use there, it is often highly customized by themes, which causes Cart/Checkout to break when theme styles leak into it. Ideally, we should either be super aggressive there, or switch to div/span (not anchor) with a role=button, this should shield us from those issues.
Example is Astra.

https://github.com/user-attachments/assets/29a9d4f1-39fc-40dc-bb4d-4cb25c4f9156

### Steps to reproduce

1. Visit [this demo site.](https://nielsendemo.mystagingwebsite.com/shop/)
2. Add item to Cart.
3. Go to Cart/Checkout and notice the broken expands.

### Expected behavior

Section heads shouldn't act/show like buttons.

### PR DESCRIPTION
**Pull request by senadir**
This PR switches some of our interactive elements in Checkout to use an Ariakit Button element (rendered as div), this prevents breakage by aggressive styles coming from theme.

The PR changes:

- Panel titles (Order Summary, Add a coupon).
- Shipping vs Local Pickup Selectors.
- Address line 2 toggle.
- Edit address toggle.

Closes https://github.com/woocommerce/woocommerce/issues/50954

For the purpose of this PR, just make sure elements aren't broken and look the same.

#### Before:

Screenshot 2024-09-13 at 11 33 30 AM
Screenshot 2024-09-13 at 11 33 12 AM

#### After:

image
Screenshot 2024-09-13 at 12 04 32 PM

#### Before:

Screenshot 2024-09-12 at 5 43 35 PM

#### After:

Screenshot 2024-09-12 at 6 48 05 PM

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Select a theme (storefront) and test panels titles, local pickup vs shipping selectors, address line 2 toggle, and address edit button.
2. Make sure they're interactive (cursor is pointer).
3. Make sure you can select them with keyboard and focus them.
4. Make sure you can toggle them using keyboard.
5. Go the Checkout edit page, select local pickup, enable price and disable icon.
6. Visit frontend, make sure they're shown on separate lines.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

Harden styles for interactive elements in Checkout block to prevent style leakage.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/packages/components/panel/index.tsx [hunk 1]
```diff
@@ -36,26 +41,31 @@ const Panel = ( {
 		[ isOpen, setIsOpen ] = state;
 	}
 
+	if ( titleTag ) {
+		deprecated( "Panel component's titleTag prop", {
+			since: '9.4.0',
+		} );
+	}
```
Comment (senadir) [inline plugins/woocommerce-blocks/packages/components/panel/index.tsx:48]: This caused an extra div that I wasn't sure why for.

File: plugins/woocommerce-blocks/package.json [hunk 2]
```diff
@@ -256,7 +256,7 @@
 		"pnpm": "9.1.3"
 	},
 	"dependencies": {
-		"@ariakit/react": "^0.4.4",
+		"@ariakit/react": "^0.4.5",
```
Comment (senadir) [inline plugins/woocommerce-blocks/package.json:259]: probably wasn't needed but still.

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx [hunk 3]
```diff
@@ -50,7 +51,8 @@ const AddressLine2Field = < T extends AddressFormValues | ContactFormValues >( {
 				/>
 			) : (
 				<>
-					<button
```
Comment (alexflorisca) [inline plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx:53]: Why are we rendering the "Add" button as a span and the "Edit" button as a div?

**Review by alexflorisca — APPROVED** [review]
This works great, tests well, thanks for working on this. Just a comment about span vs div.

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx (hunk 3 above)
Comment (senadir) [reply plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx:53]: good question, I don't have a strong preference, i think both are fine. I will use span in both.

**Pull request merged**